### PR TITLE
Fix transforms 

### PR DESCRIPTION
**Pull request by Saadnajmi**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

This picks up where #280 and #460 left off, fixing animations on macOS.

## Changelog

[macOS] [Fixed] - Fixed 3D Transforms not working properly

## Test Plan

RN-Tester's transform page look identical on both iOS and macOS

Screen Shot 2022-02-25 at 12 24 55 PM

### CODE REVIEW

File: React/Views/RCTView.m [hunk 1]
```diff
@@ -1169,6 +1172,17 @@ - (void)displayLayer:(CALayer *)layer
   backgroundColor = _backgroundColor.CGColor;
 #endif
 
+  #if TARGET_OS_OSX // [TODO(macOS ISS#2323203)
+  CATransform3D transform = self.transform3D;
```
Comment (harrieshin) [inline React/Views/RCTView.m:1176]: i am bit of a dot notation hater

File: packages/rn-tester/RNTesterPods.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1869,7 +1869,7 @@
 				ENABLE_BITCODE = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
 				ENABLE_TESTABILITY = YES;
-				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "";
+				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "arm64 ";
```
Comment (harrieshin) [inline packages/rn-tester/RNTesterPods.xcodeproj/project.pbxproj:1872]: did you remove it last time? whats going on?

File: React/Views/RCTView.m [hunk 3]
```diff
@@ -1169,6 +1172,17 @@ - (void)displayLayer:(CALayer *)layer
...
+  if (layer.anchorPoint.x == 0 && layer.anchorPoint.y == 0 && !CATransform3DEqualToTransform(transform, CATransform3DIdentity)) {
+    // This compensates for the fact that layer.anchorPoint is {0, 0} instead of {0.5, 0.5} on macOS for some reason.
```
Comment (harrieshin) [inline React/Views/RCTView.m:1178]: seems like we should add https://developer.apple.com/documentation/quartzcore/calayer/1410817-anchorpoint in  comment?

**Review by harrieshin — APPROVED** [review]

File: packages/rn-tester/RNTesterPods.xcodeproj/project.pbxproj (hunk 2 above)
Comment (Saadnajmi) [reply packages/rn-tester/RNTesterPods.xcodeproj/project.pbxproj:1872]: I think the last time I touched this file was on 0.66-stable, and not main? 
It keeps on showing up so I thought might as well commit it

File: React/Views/RCTView.m (hunk 3 above)
Comment (Saadnajmi) [reply React/Views/RCTView.m:1178]: Sure. It's worth noting that there are a lot of layer properties like this one we _might_ want to special case for macOS.

https://github.com/microsoft/react-native-macos/issues/466

File: React/Views/RCTView.m (hunk 1 above)
Comment (Saadnajmi) [reply React/Views/RCTView.m:1176]: Also this is the old ISS tag. I'll update.

**Comment by jdk** [comment]
@Saadnajmi We did some testing and this looks great.

**Comment by Saadnajmi** [comment]
> @Saadnajmi We did some testing and this looks great, thank you.

Thanks! It should be included in the next patch release of rn-macOS 0.66 =)

**Pull request merged**